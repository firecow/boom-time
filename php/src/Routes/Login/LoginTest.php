<?
declare(strict_types=1);

namespace App\Routes\Login;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class LoginTest extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $superGlobals->setHTTPPostValue("username", "Firecow");
        $superGlobals->setHTTPPostValue("password", "ikeasucks");
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/login/");
        return $routeResponse->prettyPrint();
    }
}


