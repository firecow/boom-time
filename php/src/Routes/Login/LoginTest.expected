HTTP/1.1 200 Ok
Content-Length: 297
Content-Type: application/json

bearerToken=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTEyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6Ijc0NTA3NTFmMzA3YmM3NmQ5NjBmMmQ3OTJjNjZmNzIwIiwiaW1nIjoiL3Bob3RvLzViM2UwYmM1ODY0NjNkMzQ4MDAwNDhhMyIsInVuYSI6IkZpcmVjb3cifQ.Hl4JkgPsWGExHqo_8RVpKYcl0Sm8tr98qx2A6Rb5nY0
expires=Friday, 23-Feb-2018 13:02:02 UTC

{
    "jwt": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJzaW1wbGVwaHAuY29tIiwiaWF0IjoxNTE5MzA0NTEyLCJleHAiOjE1MTkzOTA5MjIsInN1YiI6Ijc0NTA3NTFmMzA3YmM3NmQ5NjBmMmQ3OTJjNjZmNzIwIiwiaW1nIjoiL3Bob3RvLzViM2UwYmM1ODY0NjNkMzQ4MDAwNDhhMyIsInVuYSI6IkZpcmVjb3cifQ.Hl4JkgPsWGExHqo_8RVpKYcl0Sm8tr98qx2A6Rb5nY0"
}